### DISCUSSION TITLE
How to specify "camera_location"

### DISCUSSION THREAD

**Discussion by glorious1** [tylermorganwall/rayshader#185] [Q&A]
Could someone please explain what format/units should be given in  `render_snapshot`'s `camera_location` option?  I tried to give it a 3-element vector x, y, elevation, with x and y in map units (long/lat) and elevation in  meters, like the elevation data, but it was ignored.  I tried an azimuth and also ignored.

**Comment by tylermorganwall** [comment]
That option only affects the camera when `software_render = TRUE`. Otherwise, it does not adjust the camera location. The actual units are in the 3D coordinate system (not lat/long). Call `render_snapshot(print_scene_info = T, software_render = T)` to see an example of the camera location used in the scene.

**Reply by jwretham** [reply]
Hi Tyler - I'm also trying to control the camera position. Following your above comment I tried the following based on one of your examples:

`
montereybay %>%
  sphere_shade() %>%
  plot_3d(montereybay,zscale = 50, water = TRUE, waterlinecolor="white")
render_snapshot(clear=TRUE, print_scene_info = T, software_render = T) `

However I get this message:

`Error in mesh$vertices[[j]] : subscript out of bounds`

Any suggestions how to get around this?

Thanks (and thanks for Rayshader - amazing package)
